### PR TITLE
RSVP with tri-state semantics (#7)

### LINKED ISSUE #7: Phase 7: RSVP with tri-state semantics
## Summary

Parents toggle attendance per kid per event, only for kids they are linked to as a guardian. Establishes the three-state RSVP model — **attending, declined, and no-response are distinct** — and the rule that RSVP never gates roster or chart placement.

## Dependencies

- **Blocked by:** #4 (Roster), #6 (Schedule)
- **Blocks:** #8 (View page — read-only chart with RSVP state)
- **Can run in parallel with:** #5 (Returning-player picker and directory)

## Scope

`src/lib/rsvp.ts`, `src/app/t/[teamId]/schedule/`

## Tasks

- [x] Create pure `src/lib/rsvp.ts` exporting `type RsvpState = "attending" | "declined" | "no-response"`
- [x] Add `deriveRsvpState(row: Rsvp | undefined): RsvpState` — no row means `no-response`, `attending: true` means `attending`, `attending: false` means `declined`
- [x] Add a helper building a `Map` for one event across a roster, defaulting every unrepresented player to `no-response`
- [x] Write co-located tests in `src/lib/rsvp.test.ts` covering all three states and the empty-roster case
- [x] Implement the RSVP toggle server action, upserting on `@@unique([eventId, playerId])`
- [x] **Restrict the toggle to players the caller guards** via `GuardianPlayer` — a parent may never RSVP for another family's kid
- [x] Support RSVPs on practices as well as games
- [x] Surface the three states in the schedule UI, with `no-response` visually distinct from `declined`
- [x] Verify: `pnpm check` (lint → typecheck → test)
- [x] Verify: `pnpm build`

## Context

- **The schema already encodes three states; only the read path collapses them.** `Rsvp.attending` is a non-null `Boolean` (`prisma/schema.prisma:222`) and the **absence of an `Rsvp` row** is the third state. Modeling this properly here is what lets #8 and #12 both behave correctly.
- **RSVP is reporting, never a gate.** A player with `no-response` or `declined` is still fully rosterable and still fully placeable in the batting order and on the diamond. Nothing in the chart authoring path (#10, #11) may filter by RSVP state, and nothing may require an RSVP before a player can be assigned.
- This module's exported type is the contract consumed by both #8 (visual state on the view page) and #12 (readiness). Getting the shape right here avoids reworking both.
- `Rsvp` is at `prisma/schema.prisma:219-231` with `@@unique([eventId, playerId])` — the upsert target.
- Guardian scoping goes through `GuardianPlayer` (`prisma/schema.prisma:99-108`). The check belongs inside the server action, after `requireTeamAccess`, because only the action knows which player row is being written.
- Keep the derivation pure and DB-free so it tests without a database; data loading belongs in a thin wrapper. Both existing modules in `src/lib/` follow this and are the pattern to copy.

### PR DESCRIPTION
**Pull request by BRCornelius**
## Summary

Lets parents toggle attendance per kid per event (games and practices). Closes #7. RSVP is reporting-only: it never gates roster placement or the chart, and the absence of an `Rsvp` row is itself the third state ("no response"), not just an empty UI default.

## Key Decisions

- **Tri-state derivation lives in a pure module** (`src/lib/rsvp.ts`): `deriveRsvpState` and `buildRsvpStateMap` take rows in and states out with no DB access, so they're unit-tested directly and the page/action stay thin. Matches the existing `readiness.ts` / `team-access.ts` pattern of pure decision function + thin data wrapper.
- **Guardian check intersects `GuardianPlayer` with `RosterEntry` on this team**, not guardianship alone. `GuardianPlayer` is global (Decision 15), so checking it by itself would let a parent RSVP a kid onto an event for a team that kid isn't even rostered on. `guardedRosteredPlayerIds(teamId, userId)` in `src/lib/rsvps.ts` does the intersection.
- **RSVP writes require only team membership (PARENT+), not COACH+** like the other schedule mutations — this is intentionally the more permissive path in `actions.ts`, since a parent reporting their own kid's attendance isn't a roster/chart change.
- **IDs are re-derived from the resolved event/entry, never trusted from the raw form field** (`requireGuardedEvent` returns the event `getEvent` resolved; the write uses `event.id`), matching the convention in the roster actions.

## What's in Scope

- `src/lib/rsvp.ts` + tests: pure `RsvpState` derivation and per-roster state map building.
- `src/lib/rsvps.ts` + tests: team-scoped `listEventRsvps`, `guardedRosteredPlayerIds` (guardian ∩ roster), `upsertRsvp`.
- `rsvpAction` in `src/app/t/[teamId]/schedule/actions.ts`: validates response, re-resolves team access + event + guardian link, upserts on the `eventId_playerId` unique constraint.
- Attendance card on the event detail page: every rostered player shows a distinct Going / Not going / No response badge; Going/Not going toggle buttons render only for players the signed-in caller guards.

### Out of Scope

- Reminders/notifications for unanswered RSVPs — not requested for this phase.
- Any change to the readiness check or the standing chart — RSVP state is read-only input there, untouched by this PR.

## Bugs Found and Fixed During Self-Review

- **Unordered attendance list**: `getRoster` has no `orderBy`, so the attendance card rendered in raw DB order while the roster page uses `sortRoster()`. Fixed by sorting with the same utility; added a regression test asserting jersey 3 < 7 < 12 render order regardless of DB return order.
- **Attendance card displaced existing feedback**: it was inserted above the error/status message block, pushing edit-form save/error feedback out of its expected spot. Moved it below.
- **Upsert used the raw form `eventId` instead of the resolved one**: contradicted the roster actions' convention of deriving the ID from the scoped resolver rather than trusting form input. Fixed to write `upsertRsvp(event.id, ...)` using the event `requireGuardedEvent` actually resolved.
- **Weak visual-distinction test**: the original test for "all three states render distinctly" only asserted two of three labels in one render. Replaced with a three-player render asserting all three labels, plus a separate test matching exact class attributes (`text-destructive` vs `text-muted-foreground`) so the assertion can't pass vacuously against a shared substring.

## Known Gaps / Follow-ups

- No live-database verification yet — this repo has no applied migrations (see `AGENTS.md`), so writes have only been exercised against mocked Prisma calls in tests.

## Test Plan

- [x] Run this repo's full check gate (see `AGENTS.md` → `## Commands`) — lint, typecheck, tests, and build must all pass. Ran `pnpm check` and `pnpm build`; 460 tests passing, both green.
- [ ] Manual: once a DB is provisioned, open an event detail page as a parent, click "Going" for a guarded kid, confirm the badge and button state update and the change survives a reload; click "Not going" and confirm the same; confirm a kid with no RSVP row shows "No response" styled distinctly from "Not going".
- [x] Edge cases checked in tests: RSVP for a player the caller doesn't guard is rejected (`?error=not-your-player`); RSVP against an event on another team redirects instead of writing; archived team rejects the write; empty roster shows a placeholder instead of an empty list.

---
_Generated by [Claude Code](https://claude.ai/code/session_016QXXcVH5Pa4HCWJ2Kv6ZEn)_

### CODE REVIEW

File: src/lib/rsvp.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/// order and on the diamond.
+///
+/// Deliberately pure and DB-free so it tests without a database. Data
+/// loading belongs in the thin wrapper next door, `src/lib/rsvps.ts`.
+
+export type RsvpState = "attending" | "declined" | "no-response";
+
+/// The one field of `Rsvp` the derivation needs — structural, so tests and
+/// callers never import the generated Prisma client.
+export type RsvpRow = { playerId: string; attending: boolean };
```
[the remark this answers is not in the document]
Comment (BRCornelius) [reply src/lib/rsvp.ts]: Good catch — the comment was written for `deriveRsvpState`'s needs (just `attending`) but the type also carries `playerId` for `buildRsvpStateMap`'s keying. Reworded to describe both fields accurately in 266675f.

---
_Generated by [Claude Code](https://claude.ai/code)_
